### PR TITLE
Add deployment/version notes to use case READMEs

### PR DESCRIPTION
**Pull request by hayesry**
*Issue #, if available:* 297

*Description of changes:*
Added additional documentation to each use case README highlighting the following:
- For all use cases, add a note informing users that they must have matching versions of AWS CDK and AWS Solutions Constructs dependencies installed - to prevent version mismatch and typing errors.
- For all use cases, add a note detailing the intended behavior of the use case construct and how it is to be used/deployed by the user.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: ded3234dfa483d7f9d712f99dc1be81110b56bd4
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6f1fa461-e6c1-4db9-aa46-b74dbd2a396e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 6cd776ee9cdec734469ec705129b58adc9fd7965
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0f6979cb-0dbc-4201-a71f-43e1d6685daf%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/use_cases/aws-custom-glue-etl/README.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 This use case implements an example using the Kinesis Data Streams Glue Job construct.
 
+> **Note:** This use case is designed to be built and deployed into your account from your local environment using the AWS CDK Toolkit (or CLI). For information on the toolkit and how to install and configure it, please see the [guide](https://docs.aws.amazon.com/cdk/latest/guide/cli.html).
+
+> **Note:** To ensure proper functionality, the AWS Solutions Constructs packages and AWS CDK packages in your project must be the same version.
```
Comment (biffgaut) [inline source/use_cases/aws-custom-glue-etl/README.md:7]: I don't think this is enough for an issue of this severity. How about:

> **IMPORTANT:** To ensure proper functionality, the AWS Solutions Constructs packages and AWS CDK packages in your project must be the same version. If you are experiencing odd behavior, check in your package.json file that all references to CDK modules and AWS Solutions Constructs modules have the same version number, and that none include the preceding caret '^'. Once the versions are aligned, delete package-lock.json and the node_modules folder and run npm install to sync the modules.

**Review by biffgaut — CHANGES_REQUESTED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: ee629ff11c03758ebafca50a720e2942770cf973
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f2ff8955-b4b5-4a25-bd7e-da42624c5594%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 511b1ed25bc6c3a53d35d15ba4e723ab3407e54f
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=817ccac1-b2cf-4c93-be52-b7fb0433024f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: fab3ac1e65e6ed8fdd989143001ca569602893b4
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7ba3a978-cc64-4044-aaef-2a3ee37211b5%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 9dc928da407a3c04235a6cebf5686813c165d14a
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2ce7c0cd-428d-4581-b676-f0734981265a%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: source/use_cases/aws-custom-glue-etl/README.md [hunk 2]
```diff
@@ -18,8 +18,13 @@ process and store it in the target datastore as defined by the ETL script (for t
 Below are the steps to deploy the use case:
 
 ```
+# Set the proper version numbers in the package.json file
+../../../deployment/align-version.sh
```
Comment (biffgaut) [inline source/use_cases/aws-custom-glue-etl/README.md:22]: Don't you need npm install after this?

Comment (hayesry) [reply source/use_cases/aws-custom-glue-etl/README.md:22]: Good catch!

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: 5749c86896282e8191fa0bec3fa28e7caad7b238
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ffe459ce-da20-4cdb-9f6a-a26b7125a2e3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Review by biffgaut — APPROVED** [review]

**Comment by aws-solutions-constructs-team** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: codebuildgithubautobuildPro-fkVQbXRiQi6A
* Commit ID: ef43dc46141856c5cba20fb036935457e22f9c77
* Result: SUCCEEDED
* [Build Logs](https://nis3u8ftb6.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=07c5721c-aa0b-4317-aa36-7e1bc0d1e489%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**